test/SlowTests/Issues/RavenDB-21381.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FastTests;
+using FastTests.Utils;
+using Raven.Client;
+using Raven.Client.Documents.Operations.Revisions;
+using Raven.Server.Documents;
+using Raven.Server.Documents.Revisions;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Utils;
+using Tests.Infrastructure;
+using Voron.Data.Tables;
+using Xunit;
+using Xunit.Abstractions;
+using static SlowTests.Issues.RavenDB_13335;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_21381 : ClusterTestBase
+    {
+        public RavenDB_21381(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [RavenTheory(RavenTestCategory.Revisions)]
+        [RavenData(DatabaseMode = RavenDatabaseMode.All)]
+        public async Task AdoptOrphanedRevisionsTest(Options options)
+        {
+            using var store = GetDocumentStore(options);
+
+            var configuration = new RevisionsConfiguration
+            {
+                Default = new RevisionsCollectionConfiguration
+                {
+                    Disabled = false,
+                    MinimumRevisionsToKeep = 100
+                }
+            };
+
+            if (options.DatabaseMode == RavenDatabaseMode.Sharded)
+            {
+                var shards = Sharding.GetShardsDocumentDatabaseInstancesFor(store);
+                await RevisionsHelper.SetupRevisionsOnShardedDatabaseAsync(store, Server.ServerStore, configuration: configuration, shards);
+            }
+            else
+            {
+                await RevisionsHelper.SetupRevisionsAsync(store, Server.ServerStore, configuration: configuration);
+            }
```

Comment:
you don't need this `if`, simply use `SetupRevisionsAsync`

Reply:
https://github.com/ravendb/ravendb/pull/17805#discussion_r1422345050

Reply:
Done